pkg/controller/instance/plan_execution_engine.go

```diff
@@ -121,15 +129,15 @@ func executePlan(plan *activePlan, metadata *executionMetadata, c client.Client,
 
 	if allPhasesCompleted {
 		log.Printf("PlanExecution: All phases on plan %s and instance %s are healthy", plan.Name, metadata.instanceName)
-		newState.State = v1alpha1.PhaseStateComplete
+		newState.Status = v1alpha1.ExecutionComplete
 	}
 
 	return newState, nil
 }
 
 func executeStep(step v1alpha1.Step, state *v1alpha1.StepStatus, resources []runtime.Object, c client.Client) error {
-	if isInProgress(state.State) {
-		state.State = v1alpha1.PhaseStateInProgress
+	if isInProgress(state.Status) {
```

Comment:
General comment: I'm wondering if instead of doing all of this inline setting of status fields everywhere, we could create a set of transformer functions that take a state and the next step, executes it, and returns the state of the PE at that point. Could make everything a lot more fluent and set the stage for various transformations.

Reply:
oh yeah, that was my original intentioon behing this executePlan refactoring but at that point I did not do it because it's hard to "compute the next step" - you need to figure out if it's update or create (that's easy) and then what to update (so compute the diff) which is harder. I think moving in this direction would make the test really testable though. Are you ok with me pursuing that direction in the next phase? as it's rather more a testability improvement in the execution engine than the Instance controller thing...

Reply:
I'm ok with pursuing this in the next phase and agree :+1: